internal/image/resource_image.go

```diff
@@ -125,16 +125,37 @@ func (r ImageResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 				},
 			},
 
-			// Computed attributes.
-
-			"resource_id": schema.StringAttribute{
+			"architecture": schema.StringAttribute{
+				Optional: true,
 				Computed: true,
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
```

Comment:
You should be able to use github.com/lxc/incus/shared/osarch and go through architectureNames instead of hardcoding the list.